markdown/stencil-docs/custom-javascript-implementations/adding-javascript.md

```diff
@@ -56,6 +58,133 @@ Many third-party JavaScript components are distributed with npm (Node Package Ma
 
 npm facilitates managing third-party JavaScript components by placing each JavaScript component – and any of its dependencies – in the correct directories. However, as a developer, you will still need to edit your theme files to wire up the JavaScript component to expose it on your storefront. You will find several examples of this on the following pages.
 
+---
+
+<a href='#adding_development-options' aria-hidden='true' class='block-anchor'  id='adding_development-options'><i aria-hidden='true' class='linkify icon'></i></a>
+
+## Page Types and Javascript API
+
+<a href='#adding_page-types' aria-hidden='true' class='block-anchor'  id='page-types_stencil-utils'><i aria-hidden='true' class='linkify icon'></i></a>
+
+Stencil themes include an API for running JavaScript on a per-page basis.
+
+To properly write JavaScript for your theme, you will have the following page types available to you:
+
+* `pages/account/addresses`
+* `pages/account/add-address`
+* `pages/account/add-return`
+* `pages/account/add-wishlist`
+* `pages/account/recent-items`
+* `pages/account/download-item`
+* `pages/account/edit`
+* `pages/account/return-saved`
+* `pages/account/returns`
+* `pages/auth/login`
+* `pages/auth/account-created`
+* `pages/auth/create-account`
+* `pages/auth/new-password`
+* `pages/blog`
+* `pages/blog-post`
+* `pages/brand`
+* `pages/brands`
+* `pages/cart`
+* `pages/category`
+* `pages/compare`
+* `pages/errors`
+* `pages/gift-certificate/purchase`
+* `pages/gift-certificate/balance`
+* `pages/gift-certificate/redeem`
+* `global`
+* `pages/home`
+* `pages/order-complete`
+* `pages/page`
+* `pages/product`
+* `pages/search`
+* `pages/sitemap`
+* `pages/subscribed`
+* `pages/account/wishlist-details`
+* `pages/account/wishlists`
+* `pages/order-confirmation`
+* `pages/checkout`
+
+These page types correspond to the pages within your theme. Each of these page types maps to an ES6 module that extends the base `PageManager` abstract class:
+
+<div class="HubBlock-header">
+    <div class="HubBlock-header-title flex items-center">
+        <div class="HubBlock-header-name">pages/blog extending the page PageManager class</div>
+    </div><div class="HubBlock-header-subtitle"></div>
+</div>
+
+<!--
+title: "pages/blog extending the page PageManager class"
+subtitle: ""
+lineNumbers: true
+-->
+
+```js
+export default class Blog extends PageManager {
+        constructor() {
+            //Setup code goes here – attach to internals, and use internals as you would 'this'
+        }
+    }
+```
+
+<div class="HubBlock--callout">
+<div class="CalloutBlock--">
+<div class="HubBlock-content">
+    
+<!-- theme:  -->
+
+> 1. Notice the page types correspond to the html pages in [cornerstone/templates/pages/](https://github.com/bigcommerce/cornerstone/tree/master/templates/pages)

```

Comment:
Remove the 1.

Reply:
<div class="CalloutBlock--"> add the callout type. 
[See Components](https://github.com/bigcommerce/dev-docs/blob/master/markdown/_components.md#callouts)